tests/should_fail/bug/695.coma

```diff
@@ -114,17 +113,15 @@ module M_695__inversed_if [#"695.rs" 7 0 7 78]
     | bb4 = s0
       [ s0 =  [ &_11 <- [%#s695] false ] s1
       | s1 = call_once {branch'0} {_11} (fun (_ret:()) ->  [ &_0 <- _ret ] s2)
-      | s2 = bb7 ]
+      | s2 = bb9 ]
     
-    | bb7 = bb8
+    | bb9 = bb10
```

Comment:
seems like this doesn't quite capture all the intended cases? seems like there's an interaction with loops potentially?

Reply:
We need to make sure the algorithm does the right thing with no-op loops, but appart from that, I don't see why there would be a problem with loops.

In particular, I think the dominance relation should be preserved, so that the weak topological ordering should not change (I think).

Reply:
Yes, this algorithm is single pass, so it may miss some redundant gotos (depending on the visiting order of the terminators: this is why loops are not correctly optimized). I'm going to implement @jhjourdan proposition below, which should handle those cases.